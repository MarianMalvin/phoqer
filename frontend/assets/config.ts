import { faChartBar } from '@fortawesome/free-regular-svg-icons/faChartBar';
import { faCommentAlt } from '@fortawesome/free-regular-svg-icons/faCommentAlt';
import { faEnvelope } from '@fortawesome/free-regular-svg-icons/faEnvelope';
import { faUser } from '@fortawesome/free-regular-svg-icons/faUser';
import { faBullhorn } from '@fortawesome/free-solid-svg-icons/faBullhorn';
import { faSlidersH } from '@fortawesome/free-solid-svg-icons/faSlidersH';
import { faUserPlus } from '@fortawesome/free-solid-svg-icons/faUserPlus';

import { ITabs, ITabsNum, Themes } from '../interfaces';
import routes from './routes';

const config = {
    baseUrl: {
        development: (version = 'v1'): string => 'http://phoqer.com/api/' + version,
        test: (version = 'v1'): string => 'http://phoqer.com/api/' + version,
        production: (version = 'v1'): string =>
            process.browser
                ? 'http://phoqer.com/api/' + version
                : version === 'v1'
                ? 'http://backend:8000/api/' + version
                : 'http://fastapi:8001',
    },
    uploadsUrl: (version = 'v1'): string => `http://phoqer.com/api/${version}/upload/`,
    host: (lang = 'pl'): string => `http://phoqer.com${lang === 'pl' ? '' : '/' + lang}`,
    img: 'http://phoqer.com',
    themes: ['blue', 'green', 'black-blue', 'black-violet'] as Themes[],
    offers: {
        grid: {
            desktop: 4,
            tablet: 3,
            smallTablet: 2,
            mobile: 1,
        },
    },
    category: {
        grid: {
            desktop: 6,
            tablet: 4,
            smallTablet: 3,
            mobile: 2,
        },
    },
    userNavLinks: (trans: (v: string) => string, userId: string, count?: ITabsNum): ITabs[] => [
        {
            id: 'my-profile',
            text: trans('my_profile'),
            link: routes.profile.public(userId),
            icon: faUser,
        },
        {
            id: 'my-offers',
            text: trans('my_offers'),
            link: routes.profile.private.my_offers(),
            icon: faBullhorn,
        },
        {
            id: 'messages',
            text: trans('messages'),
            link: routes.profile.private.messages(),
            icon: faEnvelope,
            count: count?.messages,
        },
        {
            id: 'reviews',
            text: trans('reviews'),
            link: routes.profile.private.reviews,
            icon: faCommentAlt,
            count: count?.reviews,
        },
        {
            id: 'referral',
            text: trans('invite_friends'),
            link: routes.profile.private.referral,
            icon: faUserPlus,
        },
        {
            id: 'settings',
            text: trans('settings'),
            link: routes.profile.private.settings(),
            icon: faSlidersH,
        },
        {
            id: 'analytics',
            text: trans('analytics'),
            link: routes.profile.private.analytics(),
            icon: faChartBar,
        },
    ],
    userProfileLinks: (trans: (v: string) => string, count?: ITabsNum): ITabs[] => [
        {
            id: 'my-offers',
            text: trans('my_offers'),
            link: routes.profile.private.my_offers(),
            icon: faBullhorn,
        },
        {
            id: 'messages',
            text: trans('messages'),
            link: routes.profile.private.messages(),
            icon: faEnvelope,
            count: count?.messages,
        },
        {
            id: 'reviews',
            text: trans('reviews'),
            link: routes.profile.private.reviews,
            icon: faCommentAlt,
            count: count?.reviews,
        },
        {
            id: 'referral',
            text: trans('invite_friends'),
            link: routes.profile.private.referral,
            icon: faUserPlus,
        },
        {
            id: 'settings',
            text: trans('settings'),
            link: routes.profile.private.settings(),
            icon: faSlidersH,
        },
        {
            id: 'analytics',
            text: trans('analytics'),
            link: routes.profile.private.analytics(),
            icon: faChartBar,
        },
    ],
};

export default config;
