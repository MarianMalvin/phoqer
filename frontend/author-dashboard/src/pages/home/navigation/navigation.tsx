import React from 'react';

import { PlusIcon, OrdersIcon, OffersIcon, List, ListButtonItem, ListLinkItem, ChatboxIcon } from 'phoqer';
import { chatsPage } from 'phoqer-shared';
import { useTranslation } from 'react-i18next';

import { useLinkClick } from 'src/hook/link-click.hook';

export const Navigation = (): JSX.Element => {
    const { t, i18n } = useTranslation();
    const handleClick = useLinkClick();

    return (
        <List>
            <ListLinkItem icon={<OrdersIcon />} href={`/${i18n.language}/author/orders`} onClick={handleClick}>
                {t('Rent requests')}
            </ListLinkItem>

            <ListLinkItem icon={<OffersIcon />} href={`/${i18n.language}/author/offers`} onClick={handleClick}>
                {t('Your offers')}
            </ListLinkItem>

            <ListLinkItem icon={<PlusIcon />} href={`/${i18n.language}/author/new-offer`} onClick={handleClick}>
                {t('Create new offer')}
            </ListLinkItem>

            <ListButtonItem icon={<ChatboxIcon />} onClick={() => chatsPage.open(i18n.language)}>
                {t('Open chats')}
            </ListButtonItem>
        </List>
    );
};
