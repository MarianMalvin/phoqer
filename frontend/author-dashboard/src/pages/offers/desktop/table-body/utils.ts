import { TableCellSizeEnum } from 'phoqer';

export interface HeaderType {
    key: string;
    title: string;
}

export const headerOrder: HeaderType[] = [
    { key: 'status', title: 'Status' },
    { key: 'options', title: 'Options' },
    { key: 'images', title: 'Images' },
    { key: 'title', title: 'Title' },
    { key: 'category', title: 'Category' },
    { key: 'reviews', title: 'Reviews' },
    { key: 'price', title: 'Price' },
    { key: 'sale_percentage', title: 'Sale percentage' },
    { key: 'sale_description', title: 'Sale description' },
    { key: 'description', title: 'Description' },
];

export const sizeMap: { [key: string]: TableCellSizeEnum } = {
    status: TableCellSizeEnum.Small,
    options: TableCellSizeEnum.Middle,
    images: TableCellSizeEnum.Small2Extra,
    title: TableCellSizeEnum.Large5Extra,
    description: TableCellSizeEnum.Large8Extra,
    sale_percentage: TableCellSizeEnum.MiddleExtra,
    sale_description: TableCellSizeEnum.Large4Extra,
    reviews: TableCellSizeEnum.Middle,
    price: TableCellSizeEnum.Middle,
    category: TableCellSizeEnum.Middle,
};
