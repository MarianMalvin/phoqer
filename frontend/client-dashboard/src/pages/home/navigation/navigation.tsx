import React from 'react';

import { OrdersIcon, OptionsIcon, List, ListLinkItem, ListButtonItem, ChatboxIcon } from 'phoqer';
import { chatsPage } from 'phoqer-shared';
import { useTranslation } from 'react-i18next';

import { useLinkClick } from 'src/hook/link-click.hook';

export const Navigation = (): JSX.Element => {
    const { t, i18n } = useTranslation();
    const handleClick = useLinkClick();

    return (
        <List>
            <ListLinkItem icon={<OrdersIcon />} href={`/${i18n.language}/client/orders`} onClick={handleClick}>
                {t('My orders')}
            </ListLinkItem>

            <ListLinkItem icon={<OptionsIcon />} href={`/${i18n.language}/client/settings`} onClick={handleClick}>
                {t('My settings')}
            </ListLinkItem>

            <ListButtonItem icon={<ChatboxIcon />} onClick={() => chatsPage.open(i18n.language)}>
                {t('Open chats')}
            </ListButtonItem>
        </List>
    );
};
