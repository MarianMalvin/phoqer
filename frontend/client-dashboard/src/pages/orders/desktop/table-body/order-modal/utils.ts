export const cellMap = [
    { key: 'title', title: 'Title' },
    { key: 'status', title: 'Status' },
    { key: 'date', title: 'Created' },
    { key: 'expired', title: 'Expired' },
    { key: 'price', title: 'Price' },
    { key: 'category', title: 'Category' },
    { key: 'country', title: 'Country' },
    { key: 'city', title: 'City' },
    { key: 'zip', title: 'ZIP' },
    { key: 'address', title: 'Address' },
    { key: 'comment', title: 'Comment' },
    { key: 'sale', title: 'Sale' },
    { key: 'sale_description', title: 'Sale Description' },
];
