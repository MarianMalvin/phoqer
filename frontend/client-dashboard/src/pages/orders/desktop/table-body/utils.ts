import { TableCellSizeEnum } from 'phoqer';

export interface HeaderType {
    key: string;
    title: string;
}

export const headerOrder: HeaderType[] = [
    { key: 'status', title: 'Status' },
    { key: 'options', title: 'Options' },
    { key: 'images', title: 'Images' },
    { key: 'title', title: 'Title' },
    { key: 'price', title: 'Price' },
    { key: 'user', title: 'Author' },
    { key: 'address', title: 'Address' },
    { key: 'comment', title: 'Comment' },
    { key: 'date', title: 'Created' },
    { key: 'expired', title: 'Expiration Date' },
];

export const sizeMap: { [key: string]: TableCellSizeEnum } = {
    status: TableCellSizeEnum.Small,
    options: TableCellSizeEnum.Middle,
    title: TableCellSizeEnum.Large3Extra,
    images: TableCellSizeEnum.Small2Extra,
    price: TableCellSizeEnum.Middle,
    address: TableCellSizeEnum.Large4Extra,
    comment: TableCellSizeEnum.Large3Extra,
    user: TableCellSizeEnum.LargeExtra,
    date: TableCellSizeEnum.Large2Extra,
    expired: TableCellSizeEnum.Large2Extra,
};
