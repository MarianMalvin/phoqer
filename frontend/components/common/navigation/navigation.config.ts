import { faChartBar } from '@fortawesome/free-regular-svg-icons/faChartBar';
import { faCommentAlt } from '@fortawesome/free-regular-svg-icons/faCommentAlt';
import { faEnvelope } from '@fortawesome/free-regular-svg-icons/faEnvelope';
import { faPlusSquare } from '@fortawesome/free-regular-svg-icons/faPlusSquare';
import { faUserCircle } from '@fortawesome/free-regular-svg-icons/faUserCircle';
import { faBullseye } from '@fortawesome/free-solid-svg-icons/faBullseye';
import { faSlidersH } from '@fortawesome/free-solid-svg-icons/faSlidersH';

import { ITabs, ITabsNum } from '../../../interfaces';
import routes from '../../../utils/routes';

interface IParams {
    userId?: string | number;
    count?: ITabsNum;
}

const temp: ITabsNum = {
    messages: 4,
    reviews: 5,
};

export const getBaseNavList = ({ userId, count = temp }: IParams): ITabs[] => {
    const tabs = [
        {
            id: 'my-offers',
            text: 'my_offers',
            link: routes.profile.private.my_offers(),
            icon: faBullseye,
        },
        {
            id: 'chat',
            text: 'chat',
            link: routes.profile.private.chat(),
            icon: faEnvelope,
            count: count?.messages,
        },
        {
            id: 'reviews',
            text: 'reviews',
            link: routes.profile.private.reviews,
            icon: faCommentAlt,
            count: count?.reviews,
        },
        {
            id: 'referral',
            text: 'invite_friends',
            link: routes.profile.private.referral,
            icon: faPlusSquare,
        },
        {
            id: 'settings',
            text: 'settings',
            link: routes.profile.private.settings(),
            icon: faSlidersH,
        },
        {
            id: 'analytics',
            text: 'analytics',
            link: routes.profile.private.analytics,
            icon: faChartBar,
        },
    ];

    if (userId) {
        return [
            {
                id: 'my-profile',
                text: 'my_profile',
                link: routes.profile.public(userId),
                icon: faUserCircle,
            },
            ...tabs,
        ];
    }

    return tabs;
};
