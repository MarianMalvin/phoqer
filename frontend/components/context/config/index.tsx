import Cookies from 'js-cookie';
import React, { createContext, ReactElement, useState } from 'react';

import { IConfig } from '../../../interfaces';
import notifications from '../../common/notifications';

const init: IConfig = {
    warning: true,
    hideSearchFilters: false,
    hideTopSearchQuery: false,
    hideTopOffers: false,
};
export type IConfigValue = [config: IConfig, setConfig: (value: IConfig) => void];
export const Config = createContext<IConfigValue>([init, () => undefined]);

interface IProps {
    value?: IConfig | null;
    children: JSX.Element | JSX.Element[] | ReactElement;
}

const ConfigProvider = ({ children, value }: IProps): ReactElement => {
    const [config, setConfig] = useState<IConfig>(value || init);
    const handleConfig = (data: IConfig): void => {
        try {
            Cookies.set('phoqer_config', JSON.stringify(data));
            setConfig(data);
        } catch (error) {
            notifications('error');
            setConfig(init);
        }
    };
    return <Config.Provider value={[config, handleConfig]}>{children}</Config.Provider>;
};

export default ConfigProvider;
