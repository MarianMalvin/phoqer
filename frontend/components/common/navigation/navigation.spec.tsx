import { render } from '../../../tests/tests.utils';
import Navigation from './index';
import { getBaseNavList } from './navigation.config';

describe('Navigation component', () => {
    it('matches snapshot', () => {
        const { asFragment } = render(<Navigation tabs={getBaseNavList()} />, {});
        expect(asFragment()).toMatchSnapshot();
    });

    it('matches snapshot', () => {
        const { asFragment } = render(<Navigation className="test" tabs={getBaseNavList()} />, {});
        expect(asFragment()).toMatchSnapshot();
    });
});
