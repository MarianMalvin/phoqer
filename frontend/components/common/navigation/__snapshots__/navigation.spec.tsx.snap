// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation component matches snapshot 1`] = `
<DocumentFragment>
  <ul>
    <li
      class="item-0-2-1"
    >
      <button
        class="button-0-2-2"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-bullseye fa-w-16 "
          data-icon="bullseye"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 496 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M248 8C111.03 8 0 119.03 0 256s111.03 248 248 248 248-111.03 248-248S384.97 8 248 8zm0 432c-101.69 0-184-82.29-184-184 0-101.69 82.29-184 184-184 101.69 0 184 82.29 184 184 0 101.69-82.29 184-184 184zm0-312c-70.69 0-128 57.31-128 128s57.31 128 128 128 128-57.31 128-128-57.31-128-128-128zm0 192c-35.29 0-64-28.71-64-64s28.71-64 64-64 64 28.71 64 64-28.71 64-64 64z"
            fill="currentColor"
          />
        </svg>
        <span
          class="text-0-2-3"
        >
          My offers
        </span>
      </button>
    </li>
    <li
      class="item-0-2-1"
    >
      <button
        class="button-0-2-2"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-envelope fa-w-16 "
          data-icon="envelope"
          data-prefix="far"
          focusable="false"
          role="img"
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M464 64H48C21.49 64 0 85.49 0 112v288c0 26.51 21.49 48 48 48h416c26.51 0 48-21.49 48-48V112c0-26.51-21.49-48-48-48zm0 48v40.805c-22.422 18.259-58.168 46.651-134.587 106.49-16.841 13.247-50.201 45.072-73.413 44.701-23.208.375-56.579-31.459-73.413-44.701C106.18 199.465 70.425 171.067 48 152.805V112h416zM48 400V214.398c22.914 18.251 55.409 43.862 104.938 82.646 21.857 17.205 60.134 55.186 103.062 54.955 42.717.231 80.509-37.199 103.053-54.947 49.528-38.783 82.032-64.401 104.947-82.653V400H48z"
            fill="currentColor"
          />
        </svg>
        <span
          class="text-0-2-3"
        >
          My messages
        </span>
        <span
          class="num-0-2-4"
        >
          4
        </span>
      </button>
    </li>
    <li
      class="item-0-2-1"
    >
      <button
        class="button-0-2-2"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-comment-alt fa-w-16 "
          data-icon="comment-alt"
          data-prefix="far"
          focusable="false"
          role="img"
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M448 0H64C28.7 0 0 28.7 0 64v288c0 35.3 28.7 64 64 64h96v84c0 7.1 5.8 12 12 12 2.4 0 4.9-.7 7.1-2.4L304 416h144c35.3 0 64-28.7 64-64V64c0-35.3-28.7-64-64-64zm16 352c0 8.8-7.2 16-16 16H288l-12.8 9.6L208 428v-60H64c-8.8 0-16-7.2-16-16V64c0-8.8 7.2-16 16-16h384c8.8 0 16 7.2 16 16v288z"
            fill="currentColor"
          />
        </svg>
        <span
          class="text-0-2-3"
        >
          Reviews
        </span>
        <span
          class="num-0-2-4"
        >
          5
        </span>
      </button>
    </li>
    <li
      class="item-0-2-1"
    >
      <button
        class="button-0-2-2"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-plus-square fa-w-14 "
          data-icon="plus-square"
          data-prefix="far"
          focusable="false"
          role="img"
          viewBox="0 0 448 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M352 240v32c0 6.6-5.4 12-12 12h-88v88c0 6.6-5.4 12-12 12h-32c-6.6 0-12-5.4-12-12v-88h-88c-6.6 0-12-5.4-12-12v-32c0-6.6 5.4-12 12-12h88v-88c0-6.6 5.4-12 12-12h32c6.6 0 12 5.4 12 12v88h88c6.6 0 12 5.4 12 12zm96-160v352c0 26.5-21.5 48-48 48H48c-26.5 0-48-21.5-48-48V80c0-26.5 21.5-48 48-48h352c26.5 0 48 21.5 48 48zm-48 346V86c0-3.3-2.7-6-6-6H54c-3.3 0-6 2.7-6 6v340c0 3.3 2.7 6 6 6h340c3.3 0 6-2.7 6-6z"
            fill="currentColor"
          />
        </svg>
        <span
          class="text-0-2-3"
        >
          Invite friends
        </span>
      </button>
    </li>
    <li
      class="item-0-2-1"
    >
      <button
        class="button-0-2-2"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-sliders-h fa-w-16 "
          data-icon="sliders-h"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M496 384H160v-16c0-8.8-7.2-16-16-16h-32c-8.8 0-16 7.2-16 16v16H16c-8.8 0-16 7.2-16 16v32c0 8.8 7.2 16 16 16h80v16c0 8.8 7.2 16 16 16h32c8.8 0 16-7.2 16-16v-16h336c8.8 0 16-7.2 16-16v-32c0-8.8-7.2-16-16-16zm0-160h-80v-16c0-8.8-7.2-16-16-16h-32c-8.8 0-16 7.2-16 16v16H16c-8.8 0-16 7.2-16 16v32c0 8.8 7.2 16 16 16h336v16c0 8.8 7.2 16 16 16h32c8.8 0 16-7.2 16-16v-16h80c8.8 0 16-7.2 16-16v-32c0-8.8-7.2-16-16-16zm0-160H288V48c0-8.8-7.2-16-16-16h-32c-8.8 0-16 7.2-16 16v16H16C7.2 64 0 71.2 0 80v32c0 8.8 7.2 16 16 16h208v16c0 8.8 7.2 16 16 16h32c8.8 0 16-7.2 16-16v-16h208c8.8 0 16-7.2 16-16V80c0-8.8-7.2-16-16-16z"
            fill="currentColor"
          />
        </svg>
        <span
          class="text-0-2-3"
        >
          Settings
        </span>
      </button>
    </li>
    <li
      class="item-0-2-1"
    >
      <button
        class="button-0-2-2"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-chart-bar fa-w-16 "
          data-icon="chart-bar"
          data-prefix="far"
          focusable="false"
          role="img"
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M396.8 352h22.4c6.4 0 12.8-6.4 12.8-12.8V108.8c0-6.4-6.4-12.8-12.8-12.8h-22.4c-6.4 0-12.8 6.4-12.8 12.8v230.4c0 6.4 6.4 12.8 12.8 12.8zm-192 0h22.4c6.4 0 12.8-6.4 12.8-12.8V140.8c0-6.4-6.4-12.8-12.8-12.8h-22.4c-6.4 0-12.8 6.4-12.8 12.8v198.4c0 6.4 6.4 12.8 12.8 12.8zm96 0h22.4c6.4 0 12.8-6.4 12.8-12.8V204.8c0-6.4-6.4-12.8-12.8-12.8h-22.4c-6.4 0-12.8 6.4-12.8 12.8v134.4c0 6.4 6.4 12.8 12.8 12.8zM496 400H48V80c0-8.84-7.16-16-16-16H16C7.16 64 0 71.16 0 80v336c0 17.67 14.33 32 32 32h464c8.84 0 16-7.16 16-16v-16c0-8.84-7.16-16-16-16zm-387.2-48h22.4c6.4 0 12.8-6.4 12.8-12.8v-70.4c0-6.4-6.4-12.8-12.8-12.8h-22.4c-6.4 0-12.8 6.4-12.8 12.8v70.4c0 6.4 6.4 12.8 12.8 12.8z"
            fill="currentColor"
          />
        </svg>
        <span
          class="text-0-2-3"
        >
          analytics
        </span>
      </button>
    </li>
  </ul>
</DocumentFragment>
`;

exports[`Navigation component matches snapshot 2`] = `
<DocumentFragment>
  <ul
    class="test"
  >
    <li
      class="item-0-2-5"
    >
      <button
        class="button-0-2-6"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-bullseye fa-w-16 "
          data-icon="bullseye"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 496 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M248 8C111.03 8 0 119.03 0 256s111.03 248 248 248 248-111.03 248-248S384.97 8 248 8zm0 432c-101.69 0-184-82.29-184-184 0-101.69 82.29-184 184-184 101.69 0 184 82.29 184 184 0 101.69-82.29 184-184 184zm0-312c-70.69 0-128 57.31-128 128s57.31 128 128 128 128-57.31 128-128-57.31-128-128-128zm0 192c-35.29 0-64-28.71-64-64s28.71-64 64-64 64 28.71 64 64-28.71 64-64 64z"
            fill="currentColor"
          />
        </svg>
        <span
          class="text-0-2-7"
        >
          My offers
        </span>
      </button>
    </li>
    <li
      class="item-0-2-5"
    >
      <button
        class="button-0-2-6"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-envelope fa-w-16 "
          data-icon="envelope"
          data-prefix="far"
          focusable="false"
          role="img"
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M464 64H48C21.49 64 0 85.49 0 112v288c0 26.51 21.49 48 48 48h416c26.51 0 48-21.49 48-48V112c0-26.51-21.49-48-48-48zm0 48v40.805c-22.422 18.259-58.168 46.651-134.587 106.49-16.841 13.247-50.201 45.072-73.413 44.701-23.208.375-56.579-31.459-73.413-44.701C106.18 199.465 70.425 171.067 48 152.805V112h416zM48 400V214.398c22.914 18.251 55.409 43.862 104.938 82.646 21.857 17.205 60.134 55.186 103.062 54.955 42.717.231 80.509-37.199 103.053-54.947 49.528-38.783 82.032-64.401 104.947-82.653V400H48z"
            fill="currentColor"
          />
        </svg>
        <span
          class="text-0-2-7"
        >
          My messages
        </span>
        <span
          class="num-0-2-8"
        >
          4
        </span>
      </button>
    </li>
    <li
      class="item-0-2-5"
    >
      <button
        class="button-0-2-6"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-comment-alt fa-w-16 "
          data-icon="comment-alt"
          data-prefix="far"
          focusable="false"
          role="img"
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M448 0H64C28.7 0 0 28.7 0 64v288c0 35.3 28.7 64 64 64h96v84c0 7.1 5.8 12 12 12 2.4 0 4.9-.7 7.1-2.4L304 416h144c35.3 0 64-28.7 64-64V64c0-35.3-28.7-64-64-64zm16 352c0 8.8-7.2 16-16 16H288l-12.8 9.6L208 428v-60H64c-8.8 0-16-7.2-16-16V64c0-8.8 7.2-16 16-16h384c8.8 0 16 7.2 16 16v288z"
            fill="currentColor"
          />
        </svg>
        <span
          class="text-0-2-7"
        >
          Reviews
        </span>
        <span
          class="num-0-2-8"
        >
          5
        </span>
      </button>
    </li>
    <li
      class="item-0-2-5"
    >
      <button
        class="button-0-2-6"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-plus-square fa-w-14 "
          data-icon="plus-square"
          data-prefix="far"
          focusable="false"
          role="img"
          viewBox="0 0 448 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M352 240v32c0 6.6-5.4 12-12 12h-88v88c0 6.6-5.4 12-12 12h-32c-6.6 0-12-5.4-12-12v-88h-88c-6.6 0-12-5.4-12-12v-32c0-6.6 5.4-12 12-12h88v-88c0-6.6 5.4-12 12-12h32c6.6 0 12 5.4 12 12v88h88c6.6 0 12 5.4 12 12zm96-160v352c0 26.5-21.5 48-48 48H48c-26.5 0-48-21.5-48-48V80c0-26.5 21.5-48 48-48h352c26.5 0 48 21.5 48 48zm-48 346V86c0-3.3-2.7-6-6-6H54c-3.3 0-6 2.7-6 6v340c0 3.3 2.7 6 6 6h340c3.3 0 6-2.7 6-6z"
            fill="currentColor"
          />
        </svg>
        <span
          class="text-0-2-7"
        >
          Invite friends
        </span>
      </button>
    </li>
    <li
      class="item-0-2-5"
    >
      <button
        class="button-0-2-6"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-sliders-h fa-w-16 "
          data-icon="sliders-h"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M496 384H160v-16c0-8.8-7.2-16-16-16h-32c-8.8 0-16 7.2-16 16v16H16c-8.8 0-16 7.2-16 16v32c0 8.8 7.2 16 16 16h80v16c0 8.8 7.2 16 16 16h32c8.8 0 16-7.2 16-16v-16h336c8.8 0 16-7.2 16-16v-32c0-8.8-7.2-16-16-16zm0-160h-80v-16c0-8.8-7.2-16-16-16h-32c-8.8 0-16 7.2-16 16v16H16c-8.8 0-16 7.2-16 16v32c0 8.8 7.2 16 16 16h336v16c0 8.8 7.2 16 16 16h32c8.8 0 16-7.2 16-16v-16h80c8.8 0 16-7.2 16-16v-32c0-8.8-7.2-16-16-16zm0-160H288V48c0-8.8-7.2-16-16-16h-32c-8.8 0-16 7.2-16 16v16H16C7.2 64 0 71.2 0 80v32c0 8.8 7.2 16 16 16h208v16c0 8.8 7.2 16 16 16h32c8.8 0 16-7.2 16-16v-16h208c8.8 0 16-7.2 16-16V80c0-8.8-7.2-16-16-16z"
            fill="currentColor"
          />
        </svg>
        <span
          class="text-0-2-7"
        >
          Settings
        </span>
      </button>
    </li>
    <li
      class="item-0-2-5"
    >
      <button
        class="button-0-2-6"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-chart-bar fa-w-16 "
          data-icon="chart-bar"
          data-prefix="far"
          focusable="false"
          role="img"
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M396.8 352h22.4c6.4 0 12.8-6.4 12.8-12.8V108.8c0-6.4-6.4-12.8-12.8-12.8h-22.4c-6.4 0-12.8 6.4-12.8 12.8v230.4c0 6.4 6.4 12.8 12.8 12.8zm-192 0h22.4c6.4 0 12.8-6.4 12.8-12.8V140.8c0-6.4-6.4-12.8-12.8-12.8h-22.4c-6.4 0-12.8 6.4-12.8 12.8v198.4c0 6.4 6.4 12.8 12.8 12.8zm96 0h22.4c6.4 0 12.8-6.4 12.8-12.8V204.8c0-6.4-6.4-12.8-12.8-12.8h-22.4c-6.4 0-12.8 6.4-12.8 12.8v134.4c0 6.4 6.4 12.8 12.8 12.8zM496 400H48V80c0-8.84-7.16-16-16-16H16C7.16 64 0 71.16 0 80v336c0 17.67 14.33 32 32 32h464c8.84 0 16-7.16 16-16v-16c0-8.84-7.16-16-16-16zm-387.2-48h22.4c6.4 0 12.8-6.4 12.8-12.8v-70.4c0-6.4-6.4-12.8-12.8-12.8h-22.4c-6.4 0-12.8 6.4-12.8 12.8v70.4c0 6.4 6.4 12.8 12.8 12.8z"
            fill="currentColor"
          />
        </svg>
        <span
          class="text-0-2-7"
        >
          analytics
        </span>
      </button>
    </li>
  </ul>
</DocumentFragment>
`;
